asapdiscovery-ml/asapdiscovery/ml/weights/weights.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from pathlib import Path
+from typing import List, Union  # noqa: F401
+
+import requests
+import validators
+import yaml
+
+
+def fetch_weights_from_spec(
+    yamlfile: str,
+    models: Union[list[str], str],
+    local_path: str = "./_weights/",
+    force_fetch: bool = False,
+):
+    """Fetch weights from yaml spec file.
+
+    Parameters
+    ----------
+    yamlfile : str
+        Path to yaml spec file.
+    models : List[str]
+        Model names to fetch weights for.
+    local_path : str, optional
```

Comment:
not sure if this is the correct way to do things, but I like to add the default here as well, eg
```suggestion
    local_path : str, default="./_weights/"
```